File: stack.py
stack=[]
def push():
 e=input("Enter an element")
 stack.append(e)
 print(stack)

def pop():
 if not stack:
 print("Stack is empty")
 else:
 e=stack.pop()
 print("Removed element is:",e)
 print(stack)

def peek():
 print(stack[-1])
 
while True:
 print("Selct operation 1.push 2.pop 3.peek 4.quit")
 choice=int(input())
 if choice==1:
 push()
 elif choice==2:
 pop()
 elif choice==3:
 peek()
 elif choice==4:
 break
 else:
 print("Enter correct operation")

File: stack.c
#include<stadio.h>
#include<stdlib.h>
#define N 10
int stack[N];
int top=-1
void push();
void pop();
void peek();
void display();
void main()
{
  int ch;
  while(1)
  {
    printf("Enter choice:");
    scanf("%d",&ch);
    switch(ch)
    {
      case 1:push();
             break;
      case 2:pop();
             break;
      case 3:peek();
             break;
      case 4:display();
             break;
      case 5:exit(0);
      default:printf("Invalid case\n");
    }
  }
}
void push()
{
  int ele;
  printf("Enter element:\n");
  scanf("%d",&ele);
  if(top==N-1)
  {
    printf("Stack id full\n");
  }
  else
  {
    top++;
    stack[top]=ele;
  }
}
void pop()
{
  if(top==-1)
  {
    printf("Stack is empty\n");
  }
  else
  {
    int item;
    item=stack[top];
    printf("Popped element is %d:",item);
    top--;
  }
}
void peek()
{
  if(top==-1)
  {
    printf("stack is empty\n");
  }
  else
  {
    printf("Peek element is:%d",stack[top]);
  }
}
void display()
{
  if(top==-1)
  {
    printf("Stack is empty");
  }
  else
  {
    for(int i=top;i>=0;i--)
      printf("%d",stack[i]);
  }
}      
